Apiml/OIDC authentication (#2793)
zowe/docs-site: docs/extend/extend-apiml/zowe-api-mediation-layer-security-overview.md

- **TLS client certificates**
    - Certificates are used for service-only requests

- **OIDC authentication**
 

>   API ML is now able to authenticate mainframe users with external/distributed IDP (OIDC provider), such as OKTA, KeyCloak and others.

Reviewer: ```suggestion
  API ML can authenticate mainframe users with external/distributed IDP (OIDC provider), such as OKTA, KeyCloak and others.
```

Reviewer 2: the other document refers to it as IAM system

Author: Changed to "can be configured to authenticate ..."

>   Client applications can ask their users to log in at OIDC provider's authentication page, and then access APIs with the JWT provided by the external IDP. The following process outlines that basic flow:

Reviewer 2: same question about whether it's IAM / IDP, are these interchangeable?

Author: IAM was removed from the article, as we don't prereq full-blown Identity and Access Management. It is enough to have  OAuth2 Identity Provider, preferably with OIDC authentication layer capabilities. In the article, I use intensively OIDC instead of OAuth2. This probably deserves a revision to decide if to change it OIDC word for OAuth2 and state it precisely.

  - The client application intiates OIDC authentication flow with the distributed OIDC provider.  
  - The user provides credentials as required at the provider's authentication page/end-point.

>   - The client application obtains authorization in the form of code that is exchanged for access JWT token (or an Identity JWT and Refresh JWT).

Author: `code` is an artefact, which is part of the OAuth2 protocol `code authorization flow`. It must be clear for people who aim for implementing OIDC/OAuth2 for their client applications. We provide links to the OIDC and OAuth2 protocols and I'd not recommend going any further with the users educating on all protocol details.
